plugins/woocommerce/src/Blocks/Utils/ProductGalleryUtils.php

```diff
@@ -15,17 +15,52 @@ class ProductGalleryUtils {
 	 * @return array An array of image data for the product gallery.
 	 */
 	public static function get_product_gallery_image_data( $product ) {
-		$image_data = array();
-		$image_ids  = self::get_product_gallery_image_ids( $product );
+		$image_data = array(
+			// Images specifically added to the gallery.
+			'gallery_images'   => array(),
+			// Images that are variations of the product. These may not exist in the gallery.
+			'variation_images' => array(),
+			// All images for the product.
+			'all_images'       => array(),
+		);
+
```

Comment:
Let's avoid doing computations if it's not a product

```suggestion
if ( ! $product instanceof \WC_Product ) {
			wc_doing_it_wrong( __FUNCTION__, __( 'Invalid product object.', 'woocommerce' ), '9.8.0' );
			return $image_data;
		}
```